solution/backend/regcore/v3views.py

```diff
@@ -0,0 +1,60 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+
+from regcore.models import Title, Part
+from regcore.views import SettingsAuthentication
+
+from regcore.serializers import (
+    ContentsSerializer,
+    TitleSerializer,
+    VersionsSerializer,
+)
+
+
+class MultipleFieldLookupMixin(object):
+    # must define lookup_fields mapping with entries like { "field_name": "url_parameter", ... }
+    def get_object(self):
+        queryset = self.get_queryset()
+        queryset = self.filter_queryset(queryset)
+        filter = {}
+        for field in self.lookup_fields:
+            if self.kwargs.get(self.lookup_fields[field], None):
+                filter[field] = self.kwargs[self.lookup_fields[field]]
+        obj = get_object_or_404(queryset, **filter)
+        self.check_object_permissions(self.request, obj)
```

Comment:
Yes, for example the TitleViewSet allows POST and PUT so we need to verify permissions

Reply:
Removed line 26 in [e534208](https://github.com/CMSgov/cmcs-eregulations/pull/373/commits/e53420886ee47f241e6f08b07bf56d2447fac532) since we're not utilizing object level permissions at the moment